src/rpc/blockchain.cpp

```diff
@@ -1345,25 +1348,25 @@ static RPCHelpMan verifychain()
     };
 }
 
-static void BuriedForkDescPushBack(UniValue& softforks, const std::string &name, int softfork_height, int tip_height) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
```

Comment:
32cf12a89619688182fcb175a63ba0356f170d9c: When removing the lock annotation here, maybe add a comment to `LOCK(cs_main)`, that it must be held while all soft forks are pushed back even though there is no annotation requiring that. Otherwise the rpc might be racy.

Alternatively keep the annotation?

Reply:
I don't think it should be racy? The versionbits structures are protected by `VersionBitsCache::m_mutex` and should be stable when provided a particular block, even if it's no longer the tip or a reorg is currently happening?

Reply:
Oh correct. `active_chain_tip` is kept unchanged while pushing back all soft forks.